crates/ra_mbe/src/syntax_bridge.rs

```diff
@@ -12,12 +12,34 @@ use tt::buffer::{Cursor, TokenBuffer};
 
 use crate::subtree_source::SubtreeTokenSource;
 use crate::ExpandError;
+use std::sync::atomic::{AtomicU32, Ordering};
 
 /// Maps `tt::TokenId` to the relative range of the original token.
-#[derive(Default)]
+#[derive(Debug, PartialEq, Eq)]
 pub struct TokenMap {
     /// Maps `tt::TokenId` to the *relative* source range.
     tokens: Vec<TextRange>,
+    map_id: u32,
+}
+
+static TOKEN_MAP_COUNTER: AtomicU32 = AtomicU32::new(0);
+
+/// Generate an unique token map id for each instance
+fn make_uniq_token_map_id() -> u32 {
+    TOKEN_MAP_COUNTER.fetch_add(1, Ordering::SeqCst)
```

Comment:
What if the counter overflows? That could cause two TokenMap's to have the same id, right?

Reply:
Yes, and I just added a panic to make it as hard error.